### PR TITLE
first draft of the infrastructure working group page

### PR DESCRIPTION
**Pull request by ChristianKuehnel**
A first draft of the website for the infrastructure working group

### CODE REVIEW

**Comment by ChristianKuehnel** [comment]
A frist draft of the website for the infrastructure working group

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md]: Tanya should have an email alias setup in the next day or so.  We should update this with the address before we publish.

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md]:
```suggestion
   proposals they want to move forward on. The SOW will consist of a problem 
```

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md]: The numbering here seems wrong.

Also, for any work where we expect to spend money, there will need to be an RFP process prospective contractors participate in.  We have to have a documented, fair and impartial method of determining who a paid SOW will be awarded.  I don't think the details need to be fleshed out in this document.  We need to explain here that paid work will be subject to an RFP process.  As a separate document we can explain the RFP process.  This should be done by the IWG once it has been formed.

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md]:
```suggestion
* Help the Community to come to a decision on using Phabricator vs. GitHub Pull Requests for code
```

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md]: Maybe mention pre-commit/pre-merge testing here?

File: content/docs/infrastructure-wg.md
Comment (sqlbyme) [inline content/docs/infrastructure-wg.md:119]: I really like this list, thank you for putting the thought into compiling all of these points.  I wonder if this list belongs here or in another working document the IWG can work through once it is formed?

**Comment by ChristianKuehnel** [comment]
Hi @tlattner and @sqlbyme ,

Please review this proposal for the infrastructure working group website. I want to point to this page in the recruiting email for the working group.

Thx, Christian

**Comment by ChristianKuehnel** [comment]
@kwk this is a draft for the IWG website. Happy to hear your thoughts. It's mostly based on the proposal doc...

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: "Tools and technologies ~~technologies~~ currently used in the ~~Infrastracture~~ **infrastructure**"

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: "... we're recruting ple**a**se contact us at TODO!"

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: "propose a change" - While we're at it... the "propose" part is something I'd like to define a bit more. Over the past few month, I've created a complicated, yet rather easy to use, project that allows for incremental incubator projects. That means you can take a somewhat replicated environment of our current setup and explore changes. I don't want to spoil it here. I'm planning to release a video of the work and how it can be used in the upcoming week or the following. I plan on releasing it as a video because I figure that infrastructure is complicated and people may be tempted to not read but scream more about what's bothering them. With a video they can sit back and see what's potentially possible with our infrastructure and how **they can shape the future** of the LLVM developer workflow.

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: @ChristianKuehnel not sure but do you have something in mind that I began constructing back last year?

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: I would love to wait for the feedback on my video before ditching github PRs altogether. The possibilities of user-defined workflows is so much bigger with github than with phabricator.

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: I don't like the word "takeover". It bothers me because I wonder if we lose the possibility for incremental change. That's my biggest concern. I favour to use github and incrementally implement pieces of a desirable workflow there rather than defining a workflow and handing of a large chunk to a contractor. You'll probably have a hard time changing the requirements once the takeover has happened. And I'm sure that we will change our minds at one point about a specific solution. That's just normal and has happened in every other project as well.

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: With my proposal (to be demonstrated in the video to be released) we possibly don't even need extra hardware.

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: Good idea. The Rust team has some kind of deal with the github folks on github actions: https://blog.rust-lang.org/inside-rust/2020/07/23/rust-ci-is-moving-to-github-actions.html

File: content/docs/infrastructure-wg.md
Comment (kwk) [inline content/docs/infrastructure-wg.md]: I know that this is not the place to comment on this topic but in analogy you can probably only say that there needs to be lunch break but you cannot and should not tell what to eat.

**Review by kwk — CHANGES_REQUESTED** [review]
@ChristianKuehnel I have made some proposals for change and fixes some typos.

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: Sounds good,looking forward to your video 😄

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: This list is just a random collection of things I've seen or people told me about. The working group needs to sieve through this and decide where we want to start. Maybe GitHub PRs might be too big as first project...

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: We should not blindly take over existing stuff. I fully agree there. We should review and decide if it makes sense to take that over as is or shut it down or replace is with something new/better.

In all of these cases, I was in contact with the current maintainer and they would like to hand over their thing at some point in time.

Thx for pointing this out, I rephrased the section!

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: Yes! I don't really have a clear picture on what the questions we should ask, but I would like to identify the major pain points of the community and get them fixed (or less painful). This would be a nice showcase for the work of the IWG and to gain trust in the community and foundation for larger topics.

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: can't wait to see that 😄

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: Not sure if I get your analogy. I guess we need to work out a tangible proposal (maybe as video 😉 ) and discuss that with the community. That proposal must explain:
* the workflows for the user
* the checks we want to run
* the expected latency of the checks
* overriding failed checks in case of some type of emergency
* ...

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md]: > The numbering here seems wrong.

What do you mean? Because it's all "1."? The renderer will [automagically replace](https://guides.github.com/features/mastering-markdown/) these with the right numbering.

> there will need to be an RFP process

Yes, this is what I understood from our discussions. I added a sentence around the RFP process and that we need to work that out.

File: content/docs/infrastructure-wg.md
Comment (ChristianKuehnel) [reply content/docs/infrastructure-wg.md:119]: Thx, I just put it here to collect all information and give people some ideas of what they could be working on.
Once we have some task management tool (probably GitHub Issues), I would move the list over, so we can review/discuss each option individually and also track the history/thoughts/rationale/...

**Comment by ChristianKuehnel** [comment]
@kwk and @sqlbyme thx for the review!

I guess we can merge the first version so I can point to it from the IWG announcement email.
Happy to come back and do improvements as we go along.

**Comment by ChristianKuehnel** [comment]
@sqlbyme can you please merge this PR (or give me merge permissions)?

**Pull request merged**
